### PR TITLE
Features/updater refactor

### PR DESCRIPTION
**Pull request by dpopleton**
A complete refactor of the updater. New logic has been implemented and the code is more readable and robust. It did come at a cost of run time, with the previous instance taking milliseconds and this instance taking seconds.
Nonetheless, the whole thing should function identically to the previous version with the additional logic steps added.
In addition a function and method was added to the api. More will need to be added to datasets in the future.

This has been fully tested with local databases, however the tests may fail currently. Hopefully those are fixed before I leave.

### CODE REVIEW

**Comment by dpopleton** [comment]
The tests currently fail due to the taxids not being in the taxonomy database.
Previously the taxonomy database was a truncated version of the whole database with a few extra datapoints.
For the updater, I am testing with made up data and the taxids don't exist.
I would prefer to change the database, but I don't know how.

As is, a manual set of tests with several cores has succeeded without issue.

File: src/ensembl/production/metadata/updater/core.py [hunk 1]
```diff
@@ -58,496 +62,411 @@ def process_core(self, **kwargs):
...
+
+                # Create an AssemblySequence object.
+                assembly_sequence = AssemblySequence(
+                    name=name,
+                    assembly=assembly,
+                    accession=accession,
+                    chromosomal=info["chromosomal"],
+                    length=info["length"],
+                    sequence_location=info["location"],
+                    # sequence_checksum="", Not implemented
```
Comment (andrewyatz) [inline src/ensembl/production/metadata/updater/core.py:330]: Checksums will be in the core databases now so we can accept and load this right away correct?

**Review by andrewyatz — CHANGES_REQUESTED** [review]
See request to load checksums into the schema from the core db

File: src/ensembl/production/metadata/updater/core.py (hunk 1 above)
Comment (ens-st3) [reply src/ensembl/production/metadata/updater/core.py:330]: Should we remove the columns from the schema ?

Comment (ens-st3) [reply src/ensembl/production/metadata/updater/core.py:330]: After more discussions yesterday the answer is no. Reason - the genome browser may use these data to retrieve CHECKSUMS in order to request sequence from refget proxy.

Comment (ens-st3) [reply src/ensembl/production/metadata/updater/core.py:330]: Also, the metadata API will eventually be the source of truth of these CHECKSUMS for programmatic access. We definitely need it now

Comment (dpopleton) [reply src/ensembl/production/metadata/updater/core.py:330]: Will not do, as the checksums will be generated post-handover

**Review by andrewyatz — APPROVED** [review]
Hard to assess full impact but looks good from what I can understand

**Comment by vinay-ebi** [comment]
Are we including code to exclude LRG regions in the core update module?

**Pull request merged**